ansible/config_sonic_basedon_testbed.yml

```diff
@@ -75,14 +75,37 @@
     delegate_to: localhost
 
   - name: find interface name mapping and individual interface speed if defined from dut
-    port_alias: hwsku="{{ hwsku }}"
+    port_alias:
+      hwsku: "{{ hwsku }}"
+      card_type: "{{ card_type | default('fixed') }}"
+      hostname: "{{ inventory_hostname | default('') }}"
+      start_switchid: "{{ start_switchid | default(0) }}"
     when: deploy is defined and deploy|bool == true
 
   - name: find interface name mapping and individual interface speed if defined with local data
-    port_alias: hwsku="{{ hwsku }}" num_asic="{{ num_asics }}"
+    port_alias:
+      hwsku: "{{ hwsku }}"
+      num_asic: "{{ num_asics }}"
+      card_type: "{{ card_type | default('fixed') }}"
+      hostname: "{{ inventory_hostname | default('') }}"
+      start_switchid: "{{ start_switchid | default(0) }}"
     delegate_to: localhost
     when: deploy is not defined or deploy|bool == false
 
+  - name: find and generate fabric ASIC infomation
+    fabric_info:
+      hwsku: "{{ hwsku }}"
+      num_fabric_asic: "{{ num_fabric_asics | default(0) }}"
+      asics_host_pfx: "{{ asics_host_ip | default(None) }}"
+      asics_host_pfx6: "{{ asics_host_ipv6 | default(None) }}"
+    when: deploy is defined and deploy|bool == true
```

Comment:
if deploy is not defined or is false, will need fabric_info to generate minigraph, how will that scenario look like?

Reply:
Is there a need to have fabric_info when deploy is false ? 
then i can remove the when clause. 

I followed what was done for gen-mg of linecard or fixed systems where some steps like finnd interface name mapping, speed use the same deploy == true check.
  - name: find interface name mapping and individual interface speed if defined from dut

Reply:
My understanding is that we should be able to run gen-mg for any topology.
Doing a gen-mg or deploy-mg should result in the same minigraph. Was this something that you were able to test?
Going by this point, yes, we should be able to run fabric_info when deploy is false; which is for gen-mg scenario.
name: find interface name mapping and individual interface speed if defined from dut - this step also has 2 scenarios, one when deploy is false and another when deploy is true.

Reply:
ok, will remove the restrictions of deploy == true